crates/krator/src/manager/watch.rs

```diff
@@ -0,0 +1,49 @@
+use kube::{
+    api::{DynamicObject, GroupVersionKind, ListParams},
+    Resource,
+};
+use kube_runtime::watcher::Event;
+
+/// Captures configuration needed to configure a watcher.
+#[derive(Clone, Debug)]
+pub struct Watch {
+    /// The (group, version, kind) tuple of the resource to be watched.
+    pub gvk: GroupVersionKind,
+    /// Optionally restrict watching to namespace.
+    pub namespace: Option<String>,
+    /// Restrict to objects matching list params (default watches everything).
+    pub list_params: ListParams,
+}
+
+impl Watch {
+    pub fn new<
+        R: Resource<DynamicType = ()> + serde::de::DeserializeOwned + Clone + Send + 'static,
+    >(
+        namespace: Option<String>,
+        list_params: ListParams,
+    ) -> Self {
+        let gvk = GroupVersionKind::gvk(&R::group(&()), &R::version(&()), &R::kind(&())).unwrap();
```

Comment:
This only errors if `version` or `kind` are empty strings. I think since we are getting those values from a type that implements `Resource`, and not just arbitrary input, its reasonable to expect them to be well-formed. Thoughts?

Reply:
Only if it is an unrecoverable error. I can just imagine a case where there could have just been data corruption or something else on k8s side that could cause an error that doesn't necessarily need to panic, but I'll leave it up to you

Reply:
Did we come to a consensus on this answer? Just re-checking as I noticed @thomastaylor312 approved the PR

Reply:
I'd like to leave this for now. I don't see a scenario where this would be malformed and also something we could recover from.

Reply:
Actually, the real answer here is that I'm opening a PR for `kube` 0.54 in a few minutes, and this method no longer returns a result starting in 0.53.